nms/app/views/domain-proxy/LogsList.js

```diff
@@ -0,0 +1,418 @@
+    width: theme.spacing(19),
+  },
+  filterText: {
+    color: colors.primary.comet,
+  },
+  filterActionsWrapper: {
+    paddingRight: theme.spacing(4),
+  },
+  filterActions: {
+    textAlign: 'right',
+  },
+}));
+
+type LogRowType = {
+  body: string,
+  fccId: string,
+  serialNumber: string,
+  time: string,
+  type: 'registrationRequest' | 'spectrumInquiryRequest' | 'grantRequest' | 'heartbeatRequest' |
+  'relinquishmentRequest' | 'deregistrationRequest' | 'registrationResponse' |
+  'spectrumInquiryResponse' | 'grantResponse' | 'heartbeatResponse' |
+  'relinquishmentResponse' | 'deregistrationResponse'
+};
+
+type LogsDirectionNullable = 'SAS' | 'DP' | 'CBSD' | null;
+
+type LogsDirectionFilterProps = {
+  value: LogsDirectionNullable,
+  onChange: (value: LogsDirectionNullable) => void,
+};
+
+function LogsDirectionFilter({ value, onChange }: LogsDirectionFilterProps) {
+  const classes = useStyles();
+
+  let parsedValue;
+  switch (value) {
+    case 'SAS':
+      parsedValue = 1;
+      break;
+    case 'DP':
+      parsedValue = 2;
+      break;
+    case 'CBSD':
+      parsedValue = 3;
+      break;
+    default:
+      parsedValue = 0;
+      break;
+  }
+
+  return (
+    <FormControl>
+      <Select
+        className={classes.logsDirectionFilter}
+        value={parsedValue}
+        onChange={({ target }) => {
+          let newValue;
+          switch (parseInt(target.value)) {
+            case 1:
+              newValue = 'SAS';
+              break;
+            case 2:
+              newValue = 'DP';
+              break;
+            case 3:
+              newValue = 'CBSD';
+              break;
+            default:
+              newValue = null;
+              break;
+          }
+          onChange(newValue)
+        }}
+        input={<OutlinedInput />}
+      >
+        <MenuItem value={0}>Any</MenuItem>
+        <MenuItem value={1}>SAS</MenuItem>
+        <MenuItem value={2}>DP</MenuItem>
+        <MenuItem value={3}>CBSD</MenuItem>
+      </Select>
+    </FormControl>
+  );
+}
+
+function LogsList(props: WithAlert) {
+  const classes = useStyles();
+
+  const params = useParams();
+  const networkId: string = nullthrows(params.networkId);
+
+  const tableRef = useRef(null);
+
+  const [isAutoRefreshing, setIsAutoRefreshing] = useState(true);
+  const { startDate, endDate, setStartDate, setEndDate } = useRefreshingDateRange(
+    isAutoRefreshing,
+    REFRESH_INTERVAL,
+    () => {
+      tableRef.current && tableRef.current.onQueryChange();
+    },
+  );
+  const [serialNumber, setSerialNumber] = useState < string > ('');
+  const [fccId, setFccId] = useState < string > ('');
+  const [from, setFrom] = useState < LogsDirectionNullable > (null);
+  const [to, setTo] = useState < LogsDirectionNullable > (null);
+  const [responseCode, setResponseCode] = useState < string | null > (null);
+  const [logName, setLogName] = useState < string > ('');
+
+  const getDataFn = useCallback((query) => {
```

Comment:
Also here some test for the filtering would be nice.

Reply:
Logs filtering test added, please take a look